aws/amazon-chime-sdk-js
# Do not choose the "null" device for initial reconnection

**Issue #:** N/A

**Description of changes:**
SDK retriees to connect the session in two cases:
1. When an initial connection fails, or
2. When an ongoing meeting has a network issue or other "retryable" errors

In case 1, SDK fully disconnects the session and performs the "start" operation again. Before this PR, SDK deselects both audio and video inputs. After this PR, SDK keeps using the same devices during reconnection.

**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? By injecting a failed task during initial connection, make sure that SDK doesn't deselect audio and video inputs. After multiple retry attempts, let SDK connect the session. Ensure that the audio and video are working fine.
3. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No
4. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**From the previous PR (https://github.com/aws/amazon-chime-sdk-js/pull/699)**

> It makes sense to me. The only issue I have with this is now we need to know/remember cleanStoppedSesssionTask does not release media stream input and has to remember to call it seperately.

The name `CleanStoppedSessionTask` is a general cleaning-up task, while `LeaveAndReceiveLeaveAckTask` and `ReleaseMediaInputTask` are specific tasks. Note that `LeaveAndReceiveLeaveAckTask` was also separated from `CleanStoppedSessionTask`.

**Test steps**

### Test steps

Assume UserA and UserB are using a mobile device and MacBook, respectively.

1. Using the SDK demo, UserA joins a meeting on the mobile device.
2. UserB opens the SDK demo in MacBook. UserB types in the same meeting title and goes to the "Select Devices" page.
3. While in the "Select Devices" page, UserB disconnects WiFi.
4. UserB clicks on the "Join" button. UserB will not see UserA in the roster. Note that SDK retries to connect the session up to 2 minutes.
5. After 10 seconds, UserB connects WiFi again. After several seconds, UserB will succeed to join a meeting and view UserA.
6. UserB now speaks to the MacBook microphone. (Do not select any microphone manually in the meeting.)
7. UserA should be able to hear UserB's voice.

Note that in the current version of SDK (1.19.0), UserA cannot hear UserB's voice at step 7.
